### PR TITLE
Provide Markdown Editor to Add/Edit Description

### PR DESCRIPTION
**Pull request by hisanhunais**
## Purpose
Provide a markdown editor to add/edit the description in the publisher portal.
Related Issue: https://github.com/wso2/product-apim/issues/9272

## Approach
1. Removed the Description TextField in the Design Configurations UI page and added a link button to Add/Edit Description.
DesignConfigUI

2. Added functionality to open up a Markdown Editor to add/edit the description when the link button is clicked.
DescriptionMarkdownEditor

## Test Environment
JDK 1.8.0_241
Ubuntu 20.04.1 LTS

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/components/MarkdownEditor.jsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
+/*
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/components/MarkdownEditor.jsx:1]: @hisanhunais It seems like this is a copy of [this component](https://github.com/wso2/carbon-apimgt/blob/master/features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Documents/MarkdownEditor.jsx)
Can't we reuse it here ?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/components/MarkdownEditor.jsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+}
+
+MarkdownEditor.propTypes = {
+    api: PropTypes.shape({}).isRequired,
+    configDispatcher: PropTypes.func.isRequired,
+    classes: PropTypes.shape({}).isRequired,
+    intl: PropTypes.shape({}).isRequired,
+};
+
+export default injectIntl(withRouter(withStyles(styles)(MarkdownEditor)));
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/components/MarkdownEditor.jsx:184]: You might have taken this from an existing component, but this injects and higher-order components are a bit old with newer react hooks, Shall we replace `injectIntl` and `withRouter` , `withStyles` with their respective hooks

Comment (tmkasun) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/components/MarkdownEditor.jsx:184]: there is no usage of `withRouter` HOC , so I think we can remove that as well

**Review by tmkasun — APPROVED** [review]

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/components/MarkdownEditor.jsx (hunk 2 above)
Comment (hisanhunais) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/components/MarkdownEditor.jsx]: Fixed with https://github.com/wso2/carbon-apimgt/pull/9362/commits/f66e3a9a634e8447abebfb0698f063af3abb7f6b

**Pull request merged**